engine/src/exec.cpp

```diff
@@ -1624,11 +1642,28 @@ static bool MCPropertyParseDoubleList(MCStringRef p_input, char_t p_delimiter, u
 	uindex_t t_new_offset;
 	t_new_offset = 0;
 	
+	// PM-2015-10-13: [[ Bug 16203 ]] Replace any "empty" elements with 0.0 in the beginning of the list
+	if (MCStringGetNativeCharAtIndex(p_input, 0) == p_delimiter)
+	{
+		t_old_offset++;
+		t_new_offset++;
+		
+		t_success = t_list . Push(0.0);
+	}
+	
 	while (t_success && t_old_offset <= t_length)
```

Comment:
Looking at this loop - I think we can make this work in a slightly simpler way. The 'bug' in the current code is the t_new_offset <= t_old_offset breaking out after FirstIndexOfChar.

I think if we make the logic:
```
while(t_old_offset <= t_length)
{
  t_new_offset = firstindexofchar(delimiter, t_old_offset);
  if (t_new_offset == t_old_offset)
    Push(0)
else
   Push(attempt to convert item as number)

  advance
}
```
Then it should work?